cmd/agent/app/reporter/grpc/collector_proxy.go

```diff
@@ -41,19 +42,23 @@ func NewCollectorProxy(o *Options, mFactory metrics.Factory, logger *zap.Logger)
 	if len(o.CollectorHostPort) == 0 {
 		return nil, errors.New("could not create collector proxy, address is missing")
 	}
-	var conn *grpc.ClientConn
+	var target string
 	if len(o.CollectorHostPort) > 1 {
 		r, _ := manual.GenerateAndRegisterManualResolver()
 		var resolvedAddrs []resolver.Address
 		for _, addr := range o.CollectorHostPort {
 			resolvedAddrs = append(resolvedAddrs, resolver.Address{Addr: addr})
 		}
 		r.InitialAddrs(resolvedAddrs)
-		conn, _ = grpc.Dial(r.Scheme()+":///round_robin", grpc.WithInsecure(), grpc.WithBalancerName(roundrobin.Name))
+		target = r.Scheme() + ":///round_robin"
 	} else {
-		// It does not return error if the collector is not running
-		conn, _ = grpc.Dial(o.CollectorHostPort[0], grpc.WithInsecure(), grpc.WithBalancerName(roundrobin.Name))
+		target = o.CollectorHostPort[0]
 	}
+	// It does not return error if the collector is not running
+	conn, _ := grpc.Dial(target,
+		grpc.WithInsecure(),
+		grpc.WithBalancerName(roundrobin.Name),
+		grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor(grpc_retry.WithMax(o.MaxRetry))))
```

Comment:
How does this retry work? Is it possible to configure it with exponential backoff, or some backoff for e.g.?

Reply:
By default it is using `BackoffLinearWithJitter`. Which can be changed.

https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/retry/options.go#L26